integration-tests/tests/src/tests/shared-realms.ts

```diff
@@ -22,7 +22,57 @@ import { Realm, List } from "realm";
 import { openRealmBefore, openRealmBeforeEach } from "../hooks";
 import { createLocalConfig } from "../utils/open-realm";
 
-describe("SharedRealm operations", () => {
+describe.only("SharedRealm operations", () => {
+  describe("logger", () => {
+    it("logger callback gets called", async function () {
+      type Log = {
+        message: string;
+        level: string;
+      };
+      let logs: Log[] = [];
+
+      Realm.setLoggerCallback((level, message) => {
+        logs.push({ level, message });
+      });
+
+      Realm.setLogLevel("all");
+
+      const realm = await Realm.open({
+        schema: [{ name: "Person", properties: { name: "string" } }],
+      });
+      realm.write(() => realm.create("Person", { name: "Alice" }));
+
+      expect(logs).to.not.be.empty;
+      expect(logs.map((l) => l.level)).to.contain.members(["trace", "debug"]);
+      logs = [];
+
+      Realm.setLogLevel("trace");
+      realm.write(() => realm.create("Person", { name: "Alice" }));
+      expect(logs.map((l) => l.level)).to.contain.members(["trace", "debug"]);
+      logs = [];
+
+      Realm.setLogLevel("debug");
+      realm.write(() => realm.create("Person", { name: "Alice" }));
+      expect(logs.map((l) => l.level))
+        .to.contain("debug")
+        .and.to.not.contain("trace");
+      logs = [];
+
+      Realm.setLogLevel("warn");
+      realm.write(() => realm.create("Person", { name: "Alice" }));
+      expect(logs).to.be.empty;
+
+      Realm.setLogLevel("error");
+      realm.write(() => realm.create("Person", { name: "Alice" }));
+      expect(logs).to.be.empty;
```

Comment:
I don't think we need separate tests, otherwise we are probably verifying that the whole logger works, even at the core level. We just need to be sure the callback is called, and get a confirmation that we can change the log level. I think we can assume that core verified that all the levels work. 

Regarding "warn", "error" and so on, I was thinking about it, but it seems that those are actually raised only when using sync (like with a compensating write), and I thought it would have been too much to create a sync test just to verify the logger (also connected to what I've said before). I'm open to change my mind though 😁